docs/src/prototypes/CopyToClipboard/CopyToClipboard.tsx

```diff
@@ -0,0 +1,135 @@
+  }
+  timeoutId = undefined
+  copiedTextRef = React.createRef<HTMLElement>()
+
+  static defaultProps = {
+    timeout: 4000,
+    attached: false,
+    pointing: false,
+  }
+
+  handleClick = () => {
+    if (this.timeoutId !== undefined) {
+      clearTimeout(this.timeoutId)
+    }
+    const { timeout, value } = this.props
+    this.setState({
+      copied: true,
+    })
+    this.timeoutId = setTimeout(() => {
+      this.setState({ copied: false })
+    }, timeout)
+    copyToClipboard(value)
+  }
+
+  componentDidMount() {
+    if (this.copiedTextRef.current === null) {
+      return
+    }
+    this.setState({
+      copiedHeight: this.copiedTextRef.current.clientHeight,
+      copiedWidth: this.copiedTextRef.current.clientWidth,
+    })
+  }
+
+  render() {
+    const { attached, pointing, copyPrompt } = this.props
+    const hideContent = this.state.copied && !attached
+    const copiedStr = 'Copied to clipboard'
+    const copiedText = hideContent ? (
+      <Text size="larger" styles={{ padding: '5px 5px 5px 5px' }}>
+        {copiedStr}
+      </Text>
+    ) : (
+      <Text>{copiedStr}</Text>
+    )
+    const copyText = <Text>{copyPrompt}</Text>
+    let popupContent = this.state.copied ? copiedText : copyText
+    if (hideContent) {
+      popupContent = <Text />
+    }
+    const button = (
+      <Button
+        icon="clipboard-copied-to"
+        iconOnly
+        onClick={this.handleClick}
+        variables={siteVariables => ({
+          colorHover: !this.state.copied && siteVariables.colors.brand[400],
+        })}
+      />
+    )
+    const hiddenPopupVariables = {
+      contentBackgroundColor: 'transparent',
+    }
+    const hiddenPopupContentVariables = {
+      boxShadow: 'none',
+      borderColor: 'transparent',
+    }
+    const portalContentStyle = {
+      display: 'flex',
+      alignItems: 'center',
+      justifyContent: 'center',
+      position: 'fixed' as 'fixed',
+      backgroundColor: 'black',
+      color: 'white',
+      zIndex: 1000,
+      top: `calc(50% - ${this.state.copiedHeight / 2}px)`,
+      left: `calc(50% - ${this.state.copiedWidth / 2}px)`,
+    }
+    return (
+      <>
+        <Popup
+          pointing={pointing}
+          position="below"
+          align="center"
+          trigger={button}
+          content={{
+            content: popupContent,
+            variables: {
+              ...(hideContent && hiddenPopupContentVariables),
+            },
+          }}
+          on={['hover', 'context']}
+          variables={siteVariables => ({
+            contentColor: this.state.copied
+              ? siteVariables.colorScheme.brand.foreground4
+              : siteVariables.colors.black,
+            contentBackgroundColor: this.state.copied
+              ? siteVariables.colorScheme.brand.background4
+              : siteVariables.colors.white,
+            ...(hideContent && hiddenPopupVariables),
+          })}
+        />
+        {hideContent && (
+          <Portal
```

Comment:
Let's extract this to a separate component like `CopyButtonNotification`. We can reuse our [`createComponent()`](https://stardust-ui.github.io/react/integrate-custom-components) to host styles.

```tsx
const CopyButtonNotification = createComponent({
  displayName: 'CopyButtonNotification',
  render: () => {
    return (
      <Portal>
        <div className={classes.root}>
          <div className={classes.content}>{children}</div>
        </div>
       </Portal>
    )
  },
})
```

Reply:
Agree with createComponent, also consolidating styles in the provider or theme would be nice.

Would it be possible to create a "Notification" component instead so that you can use without the button? (which I believe Shift's example might be showing as well)

Reply:
> Would it be possible to create a "Notification" component instead so that you can use without the button?

I think so 👍